webapp/src/components/App/Singlestudy/explore/Configuration/General/dialogs/ScenarioBuilderDialog/index.tsx

```diff
@@ -99,55 +78,44 @@ function ScenarioBuilderDialog(props: Props) {
       open={open}
       onClose={onClose}
       actions={<Button onClick={onClose}>{t("button.close")}</Button>}
-      maxWidth="md"
+      maxWidth="xl"
       fullWidth
-      PaperProps={{
-        // TODO: add `maxHeight` and `fullHeight` in BasicDialog`
-        sx: { height: "calc(100% - 64px)", maxHeight: "900px" },
+      contentProps={{
+        sx: { p: 1, height: "95vh", width: 1 },
       }}
     >
-      <UsePromiseCond
-        response={res}
-        ifResolved={() => (
-          <ConfigContext.Provider value={cxValue}>
-            <Rulesets />
-            {activeRuleset && (
-              <TabContext value={currentTab}>
-                <Box sx={{ borderBottom: 1, borderColor: "divider" }}>
-                  <TabList onChange={handleTabChange}>
-                    {TABS_DATA.map(([name]) => (
-                      <Tab
-                        key={name}
-                        value={name}
-                        label={t(
-                          `study.configuration.general.mcScenarioBuilder.tab.${name}`,
-                        )}
-                      />
-                    ))}
-                  </TabList>
-                </Box>
-                {TABS_DATA.map(([name, sym]) => (
-                  <TabPanel
-                    key={name}
-                    value={name}
-                    sx={{ p: 0, pt: 2, height: 1, overflow: "auto" }}
-                  >
-                    {name === "thermal" ? (
-                      <Thermal nbYears={nbYears} />
-                    ) : (
-                      <Table
-                        nbYears={nbYears}
-                        symbol={sym}
-                        rowType={name === "ntc" ? "link" : "area"}
-                      />
-                    )}
-                  </TabPanel>
-                ))}
-              </TabContext>
-            )}
-          </ConfigContext.Provider>
-        )}
-      />
+      <ScenarioBuilderContext.Provider value={scenarioBuilderContext}>
+        <TabContext value={selectedScenario}>
```

Comment:
Use `TabsView`:

```jsx
const getTabContent = (type) => (
  <UsePromiseCond ... />
);

<TabsView
  items={SCENARIOS.map((type) => ({
    label: t(`study.configuration.general.mcScenarioBuilder.tab.${type}`),
    content: getTabContent(type),
  }))}
/>
```

Reply:
its a refactor not really an issue

Reply:
I don't say that is an issue, but If we have a common component that make the work we used it